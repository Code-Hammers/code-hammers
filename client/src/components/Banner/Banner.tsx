import React from "react";
import logo from "../../assets/hammer.png";

const Banner = (): JSX.Element => {
  return (
    <div className="bg-rose-300 p-4 md:p-6 flex items-center w-full justify-between">
      {" "}
      <img src={logo} alt="Code Hammers Logo" className="h-12 md:h-16" />
      <div className="flex-grow flex justify-center">
        <h1 className="text-2xl md:text-4xl font-semibold text-teal-600">
          Code Hammers
        </h1>{" "}
      </div>
    </div>
  );
};

export default Banner;
