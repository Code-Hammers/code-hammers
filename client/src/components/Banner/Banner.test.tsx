import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Banner from "./Banner";

xdescribe("Banner Component", () => {
  it("renders the logo image correctly", () => {
    const { getByAltText } = render(<Banner />);
    const logo = getByAltText("Code Hammers Logo") as HTMLImageElement;

    expect(logo).toBeInTheDocument();
  });

  it("renders the title text correctly", () => {
    const { getByText } = render(<Banner />);
    const title = getByText("Code Hammers");

    expect(title).toBeInTheDocument();
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<Banner />);
    expect(asFragment()).toMatchSnapshot();
  });
});
