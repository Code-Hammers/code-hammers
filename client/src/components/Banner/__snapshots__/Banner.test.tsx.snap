// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-rose-300 p-4 md:p-6 flex items-center w-full justify-between"
  >
     
    <img
      alt="Code Hammers Logo"
      class="h-12 md:h-16"
      src=""
    />
    <div
      class="flex-grow flex justify-center"
    >
      <h1
        class="text-2xl md:text-4xl font-semibold text-teal-600"
      >
        Code Hammers
      </h1>
       
    </div>
  </div>
</DocumentFragment>
`;
